docs/source/containers.rst

```diff
@@ -1,34 +1,303 @@
+Happi database, we use the concept of containers. This allows a developer to
+specify fields that are essential to every instance of a specific type.
 
-Happi allow users to extend the containers list via Python
-`entry points <https://packaging.python.org/specifications/entry-points/>`_
-specified by the **happi.containers** keyword. An example entry points can be
-found `here <https://github.com/pcdshub/pcdsdevices/blob/master/setup.py>`_.
+These fields are specified using an instance of :class:`.EntryInfo`.
+`.EntryInfo` provides several primary features:
 
-`HappiRegistry` takes care of loading the entry points and making them
-available throughout the library.
+* Mark a field as required (or optional)
+* Add default values when unspecified
+* Enforce - or validate - a certain format for the field
 
-Loading Containers
-------------------
+.. autoclass:: happi.EntryInfo
+   :members:
 
-.. autofunction:: happi.from_container
+HappiItem
+^^^^^^^^^
 
-.. autofunction:: happi.load_devices
+In order to ensure that information is entered into the database in an
+organized fashion, the client will only accept classes that inherit from
+:class:`.HappiItem`. Each item will have the key information represented as
+class attributes, available to be manipulated like any other regular property
+
+Editing the information for a container is a simple as:
+
+.. ipython:: python
+
+    from happi import HappiItem
+
+    item = HappiItem(name='my_device')
+
+    item.name = 'new_name'
+
+
+.. note::
+
+    :class:`happi.Device` class is **deprecated** due to ambiguous name,
+    conflicting with :class:`ophyd.Device`.
+
+
+Example Container
+^^^^^^^^^^^^^^^^^
+
+In order to show the flexibility of the :class:`.EntryInfo` class, we'll put
+together a new example container. The class can be invoked in the same way you
+would usually handle class inheritance, the only difference is that you specify
+class attributes as EntryInfo objects:
+
+.. ipython:: python
+
+    import re
+    from happi import HappiItem, EntryInfo
+
+    class MyItem(HappiItem):
+        """My new item, with a known model number."""
+        model_no      = EntryInfo('Model Number of Item', optional=False)
+        count         = EntryInfo('Count of Item', enforce=int, default=0)
+        choices       = EntryInfo('Choice Info', enforce=['a','b','c'])
+        no_whitespace = EntryInfo('Enforce no whitespace',
+                                   enforce = re.compile(r'[\S]*$'))
+
+By default, :class:`.EntryInfo` will create an optional init keyword argument with a
+default of ``None`` with the same name as the class attribute. A quick way
+to see how this information will be put into the the database is taking a look
+at ``dict(item)``:
+
+.. ipython:: python
+
+    item = MyItem(name="my_item", model_no="QABC1234")
+    dict(item)
+
+As shown in the example above, using the EntryInfo keywords, you can put a
+short doc string to give a better explanation of the
```

Comment:
```suggestion
short docstring to give a better explanation of the
```

Reply:
Out of curiosity, how is this docstring ever really visible?
I kind of see some `make_docstring()` and such handling stuff but I'm tired and can't find my docstring anywhere :/

Reply:
I think the answer is probably "not yet". It'd be good to generate field documentation down the line with this, similar to how we use it in ophyd/caproto documentation though.